src/crypto/crypto_cipher.cc

```diff
@@ -48,6 +48,82 @@ bool IsSupportedAuthenticatedMode(const EVP_CIPHER_CTX* ctx) {
 bool IsValidGCMTagLength(unsigned int tag_len) {
   return tag_len == 4 || tag_len == 8 || (tag_len >= 12 && tag_len <= 16);
 }
+
+// Collects and returns information on the given cipher
+void GetCipherInfo(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args[0]->IsObject());
+  Local<Object> info = args[0].As<Object>();
+
+  CHECK(args[1]->IsString() || args[1]->IsInt32());
+
+  const EVP_CIPHER* cipher;
+  if (args[1]->IsString()) {
+    Utf8Value name(env->isolate(), args[1]);
+    cipher = EVP_get_cipherbyname(*name);
+  } else {
+    int nid = args[1].As<Int32>()->Value();
+    if (nid < 0) return;
+    cipher = EVP_get_cipherbynid(nid);
+  }
+
+  if (cipher == nullptr)
+    return;
+
+  USE(info->Set(
+      env->context(),
+      env->name_string(),
+      OneByteString(env->isolate(), EVP_CIPHER_name(cipher))));
+
+  USE(info->Set(
+      env->context(),
+      FIXED_ONE_BYTE_STRING(env->isolate(), "nid"),
+      Int32::New(env->isolate(), EVP_CIPHER_nid(cipher))));
+
+  USE(info->Set(
+      env->context(),
+      FIXED_ONE_BYTE_STRING(env->isolate(), "blockSize"),
+      Int32::New(env->isolate(), EVP_CIPHER_block_size(cipher))));
+
+  USE(info->Set(
+      env->context(),
+      FIXED_ONE_BYTE_STRING(env->isolate(), "ivLength"),
+      Int32::New(env->isolate(), EVP_CIPHER_iv_length(cipher))));
+
+  USE(info->Set(
+      env->context(),
+      FIXED_ONE_BYTE_STRING(env->isolate(), "keyLength"),
+      Int32::New(env->isolate(), EVP_CIPHER_key_length(cipher))));
```

Comment:
I _think_ the same concern applies here, but I am not entirely sure. I believe some ciphers (e.g., blowfish, rc2, rc4, rc5, cast) support variable key lengths, and I assume OpenSSL will return an arbitrary default value just like it does for the IV length. If that's the case, then it should be very clear to users that the value provided by this property is not necessarily the best choice for their particular application. If different key lengths always result in different cipher names (which seems to be the case for other ciphers such as AES), this is not relevant.

Reply:
Indeed, I was investigating whether openssl provided a way of determining that and it appears there's not. We'll likely need a lookup table to augment the information returned.